File: app/src/main/java/com/sasfmlzr/findfilm/adapter/SearchAdapter.java
package com.sasfmlzr.findfilm.adapter;

import android.content.Context;
import android.database.Cursor;
import android.databinding.DataBindingUtil;
import android.support.v4.widget.CursorAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.sasfmlzr.findfilm.R;
import com.sasfmlzr.findfilm.databinding.SearchItemBinding;
import com.sasfmlzr.findfilm.fragment.discoverfilm.DiscoverFilmFragment;
import com.sasfmlzr.findfilm.request.DiscoverMovieRequest;

import java.util.List;

public class SearchAdapter extends CursorAdapter {
    private List<DiscoverMovieRequest.Result> filmList;
    private DiscoverFilmFragment.OnFilmSelectedListener filmSelectedListener;

    public SearchAdapter(Context context,
                         Cursor cursor,
                         List<DiscoverMovieRequest.Result> filmList,
                         DiscoverFilmFragment.OnFilmSelectedListener filmSelectedListener) {
        super(context, cursor, false);
        this.filmList = filmList;
        this.filmSelectedListener = filmSelectedListener;
    }

    @Override
    public void bindView(View view, Context context, Cursor cursor) {
        SearchItemBinding binding = DataBindingUtil.bind(view);
        assert binding != null;
        ViewHolder holder = new ViewHolder(binding);
        if (filmList != null) {
            holder.binding.itemSearchTitle.setText(filmList.get(cursor.getPosition()).getTitle());
        }
    }

    @Override
    public View newView(Context context, Cursor cursor, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        assert inflater != null;
        return inflater.inflate(R.layout.search_item, parent, false);
    }

    private class ViewHolder implements View.OnClickListener {
        private SearchItemBinding binding;
        final int position = getCursor().getPosition();

        ViewHolder(SearchItemBinding binding) {
            this.binding = binding;
            binding.itemSearchTitle.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            filmSelectedListener.filmClicked(filmList.get(position).getId());
        }
    }
}

File: app/src/main/java/com/sasfmlzr/findfilm/adapter/SearchAdapter.kt
package com.sasfmlzr.findfilm.adapter

import android.content.Context
import android.database.Cursor
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.cursoradapter.widget.CursorAdapter
import androidx.databinding.DataBindingUtil
import com.sasfmlzr.findfilm.R
import com.sasfmlzr.findfilm.databinding.SearchItemBinding
import com.sasfmlzr.findfilm.fragment.discoverfilm.DiscoverFilmFragment.OnFilmSelectedListener
import com.sasfmlzr.findfilm.request.DiscoverMovieRequest

class SearchAdapter(
    context: Context?,
    cursor: Cursor?,
    private val filmList: List<DiscoverMovieRequest.Result>?,
    private val filmSelectedListener: OnFilmSelectedListener
) : CursorAdapter(context, cursor, false) {
    override fun bindView(view: View, context: Context, cursor: Cursor) {
        val binding: SearchItemBinding = DataBindingUtil.bind(view)!!
        val holder = ViewHolder(binding)
        if (filmList != null) {
            holder.binding.itemSearchTitle.text = filmList[cursor.position].title
        }
    }

    override fun newView(context: Context, cursor: Cursor, parent: ViewGroup): View {
        val inflater = (context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater)
        return inflater.inflate(R.layout.search_item, parent, false)
    }

    private inner class ViewHolder(val binding: SearchItemBinding) :
        View.OnClickListener {
        val position = cursor.position
        override fun onClick(v: View) {
            filmSelectedListener.filmClicked(filmList!![position].id)
        }

        init {
            binding.itemSearchTitle.setOnClickListener(this)
        }
    }
}
